src/Symfony/Bundle/FrameworkBundle/Console/Descriptor/Descriptor.php

```diff
@@ -250,7 +255,7 @@ protected function findDefinitionsByTag(ContainerBuilder $builder, $showPrivate)
             foreach ($builder->findTaggedServiceIds($tag) as $serviceId => $attributes) {
                 $definition = $this->resolveServiceDefinition($builder, $serviceId);
 
-                if (!$definition instanceof Definition || !$showPrivate && !$definition->isPublic()) {
+                if ($showHidden xor '.' === $serviceId[0] ?? null) {
```

Comment:
Can we use parentheses here? At least I have no idea by reading the code whether `xor` takes precedence over `??` or vice versa.

Reply:
Is this actually what we want at all? Won't we skip non hidden service now if `$showHidden` is `true`?

Reply:
Can we do that in the other files too that use the same condition? :)

Reply:
never mind, I was looking at an outdated diff